new tailscale_acl fails with hujson error if acl is not known at plan time

**Describe the bug**
Can't create new `tailscale_acl` resource if acl is not known at plan time

**To Reproduce**
Steps to reproduce the behaviour:

```
provider "tailscale" {
}

terraform {
  required_providers {
    tailscale = {
      source  = "tailscale/tailscale"
      version = "~> 0.14"
    }
  }
  required_version = "~> 1.0"
}
resource "tailscale_acl" "vpn" {
  acl = jsonencode({foo=timestamp()})

  overwrite_existing_content = true
}
```

**Error**
```
│ Error: Plugin did not respond
│ 
│   with tailscale_acl.vpn,
│   on main.tf line 13, in resource "tailscale_acl" "vpn":
│   13: resource "tailscale_acl" "vpn" {
│ 
│ The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).PlanResourceChange call. The plugin logs may contain more details.
╵

Stack trace from the terraform-provider-tailscale_v0.14.0 plugin:

panic: could not parse ACL as HuJSON: hujson: line 1, column 1: invalid literal: 74D93920-ED26-11E3-AC10-0800200C9A66

goroutine 26 [running]:
github.com/tailscale/terraform-provider-tailscale/tailscale.resourceACL.func4({0xb62540?, 0xc000596660?})
	github.com/tailscale/terraform-provider-tailscale/tailscale/resource_acl.go:72 +0x9c
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.schemaMap.diffString(0xdba538?, {0xc900ac, 0x3}, 0xc000384280, 0xc00039f070, {0xdbd958, 0xc000140f80}, 0x0)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.32.0/helper/schema/schema.go:1549 +0xf8
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.schemaMap.diff(0xc0002efe30, {0xdba570, 0xc000592570}, {0xc900ac, 0x3}, 0xc000384280, 0xc000140f00, {0xdbd958, 0xc000140f80}, 0x0)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.32.0/helper/schema/schema.go:1128 +0x165
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.schemaMap.Diff(0xc0002efe30, {0xdba570, 0xc000592570}, 0xc0001324e0, 0xc000148af0, 0xce5348, {0xc71960, 0xc0003c83c0}, 0x0)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.32.0/helper/schema/schema.go:678 +0x332
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).SimpleDiff(0xdba420?, {0xdba570?, 0xc000592570?}, 0xc0001324e0, 0xc0005925a0?, {0xc71960?, 0xc0003c83c0?})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.32.0/helper/schema/resource.go:962 +0xdb
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).PlanResourceChange(0xc00037c480, {0xdba570?, 0xc0005924b0?}, 0xc000148aa0)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.32.0/helper/schema/grpc_provider.go:797 +0x996
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).PlanResourceChange(0xc000276e60, {0xdba570?, 0xc00012da70?}, 0xc0001622a0)
	github.com/hashicorp/terraform-plugin-go@v0.21.0/tfprotov5/tf5server/server.go:810 +0x3d0
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_PlanResourceChange_Handler({0xc6b840, 0xc000276e60}, {0xdba570, 0xc00012da70}, 0xc000140780, 0x0)
	github.com/hashicorp/terraform-plugin-go@v0.21.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:500 +0x1a6
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0001a1000, {0xdba570, 0xc00012d9e0}, {0xdbeaa0, 0xc0003be000}, 0xc00017d0e0, 0xc000388240, 0x12e9380, 0x0)
	google.golang.org/grpc@v1.61.0/server.go:1385 +0xdd1
google.golang.org/grpc.(*Server).handleStream(0xc0001a1000, {0xdbeaa0, 0xc0003be000}, 0xc00017d0e0)
	google.golang.org/grpc@v1.61.0/server.go:1796 +0xfb8
google.golang.org/grpc.(*Server).serveStreams.func2.1()
	google.golang.org/grpc@v1.61.0/server.go:1029 +0x8b
created by google.golang.org/grpc.(*Server).serveStreams.func2 in goroutine 12
	google.golang.org/grpc@v1.61.0/server.go:1040 +0x125

Error: The terraform-provider-tailscale_v0.14.0 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```

**Expected behaviour**
Validation doesn't fail

--- a/tailscale/resource_acl.go
+++ b/tailscale/resource_acl.go
@@ -18,6 +18,9 @@ const resourceACLDescription = `The acl resource allows you to configure a Tails
 
 If tests are defined in the ACL (the top-level "tests" section), ACL validation will occur before creation and update operations are applied.`
 
+// from https://github.com/hashicorp/terraform-plugin-sdk/blob/34d8a9ebca6bed68fddb983123d6fda72481752c/internal/configs/hcl2shim/values.go#L19
+const UnknownVariableValue = "74D93920-ED26-11E3-AC10-0800200C9A66"
+
 func resourceACL() *schema.Resource {
 	return &schema.Resource{
 		Description:   resourceACLDescription,
@@ -32,6 +35,11 @@ func resourceACL() *schema.Resource {
 		},
 		CustomizeDiff: func(ctx context.Context, rd *schema.ResourceDiff, m interface{}) error {
 			client := m.(*tailscale.Client)
+
+			//if the acl is only known after apply, then acl will be an empty string and validation will fail
+			if rd.Get("acl").(string) == "" {
+				return nil
+			}
 			return client.ValidateACL(ctx, rd.Get("acl").(string))
 		},
 		Schema: map[string]*schema.Schema{
@@ -67,6 +75,10 @@ func resourceACL() *schema.Resource {
 
 				// Use the canonical HuJSON representation of the policy in Terraform state.
 				StateFunc: func(i interface{}) string {
+					//if the acl is only known after apply, then it will be the magic UUID `UnknownVariableValue` and not valid json, and formatting will fail
+					if i.(string) == UnknownVariableValue {
+						return i.(string)
+					}
 					value, err := hujson.Format([]byte(i.(string)))
 					if err != nil {
 						panic(fmt.Errorf("could not parse ACL as HuJSON: %s", err))

--- a/tailscale/resource_acl.go
+++ b/tailscale/resource_acl.go
@@ -18,7 +18,8 @@ const resourceACLDescription = `The acl resource allows you to configure a Tails
 
 If tests are defined in the ACL (the top-level "tests" section), ACL validation will occur before creation and update operations are applied.`
 
-// from https://github.com/hashicorp/terraform-plugin-sdk/blob/34d8a9ebca6bed68fddb983123d6fda72481752c/internal/configs/hcl2shim/values.go#L19
+// From https://github.com/hashicorp/terraform-plugin-sdk/blob/34d8a9ebca6bed68fddb983123d6fda72481752c/internal/configs/hcl2shim/values.go#L19
+// TODO: use an exported variable when https://github.com/hashicorp/terraform-plugin-sdk/issues/803 has been addressed.
 const UnknownVariableValue = "74D93920-ED26-11E3-AC10-0800200C9A66"
 
 func resourceACL() *schema.Resource {

